### PR TITLE
🏢 feat: Tenant-Scoped App Config in Auth Login Flows

### PR DESCRIPTION
**Pull request by danny-avila**
## Summary

I implemented tenant-scoped `AppConfig` resolution in all auth login flows so that tenant-specific overrides (e.g., `allowedDomains`, balance settings) are applied during authentication, not just on post-login API calls.

- Add `resolveAppConfigForUser()` as a TypeScript utility in `packages/api/src/app/resolve.ts` using a DI pattern (accepts `getAppConfig` as parameter). Wraps `getAppConfig` in `tenantStorage.run()` with an `async` callback when the user has a `tenantId`, falling back to `baseOnly` for non-tenant users. JSDoc documents the dual `tenantId` propagation (ALS for DB isolation, explicit param for cache key). Auth flows apply role-level overrides only (userId deferred to post-auth).
- Apply a consistent two-phase domain check across all auth strategies (LDAP, SAML, OpenID, social login) and `requestPasswordReset`: fast-fail with the memory-cached base config (zero DB queries) to block globally denied domains, then re-check with tenant-scoped config after user lookup only when the user has a `tenantId`.
- Preserve OpenID error ordering: `AUTH_FAILED` from `findOpenIDUser` is checked before the tenant domain re-check so users with wrong providers get the correct error type. LDAP ordering change (domain check before `findUser`) is documented inline.
- Move SAML provider check before tenant config resolution to avoid unnecessary `resolveAppConfigForUser` calls for wrong-provider users.
- Prevent user-enumeration in `requestPasswordReset`: tenant domain block returns the same generic "If an account with that email exists..." response (HTTP 200) as user-not-found, so attackers cannot distinguish tenant-blocked users from non-existent ones. Phase 1 (base check, pre-`findUser`) intentionally returns Error/400 since it reveals only global domain policy, not user existence. Both phases documented in JSDoc.
- Wrap `resolveAppConfigForUser` in `requestPasswordReset` with try/catch to prevent DB errors from leaking to the client; falls back to base config on failure.
- Add audit logging (`logger.warn`) to both domain block paths in `requestPasswordReset`, consistent with all auth strategies.
- Leave `registerUser` unchanged (`baseOnly: true`) since no user identity exists yet during registration.
- Add `resolve.spec.ts` in `packages/api` with 10 tests covering baseOnly fallback, tenant-scoped resolution, real ALS context propagation (verified inside `getAppConfig` callback), undefined-role edge case, and error propagation for both paths.
- Add tenant domain rejection tests to all 4 strategy specs verifying that tenant config restrictions actually block login. Update strategy and AuthService tests to mock `resolveAppConfigForUser` via `@librechat/api`. Restore `jest.requireActual` in `socialLogin.test.js`.

## Change Type

- [x] New feature (non-breaking change which adds functionality)

## Testing

All 158 tests pass across 6 suites:

```
PASS packages/api  resolve.spec.ts           (10 tests)
PASS api/          ldapStrategy.spec.js       (8 tests)
PASS api/          socialLogin.test.js        (8 tests)
PASS api/          samlStrategy.spec.js      (26 tests)
PASS api/          openidStrategy.spec.js    (89 tests)
PASS api/          AuthService.spec.js       (17 tests)
```

### Test Configuration:
- Jest with Node test environment
- Packages built via `npm run build:data-schemas && npm run build:data-provider && npm run build:api`

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes

### CODE REVIEW

File: api/server/services/AuthService.js [hunk 1]
```diff
@@ -255,19 +256,49 @@ const registerUser = async (user, additionalData = {}) => {
...
+    } catch (err) {
+      logger.error('[requestPasswordReset] Failed to resolve tenant config, using base:', err);
+    }
+  }
+
+  if (!isEmailDomainAllowed(email, appConfig?.registration?.allowedDomains)) {
+    logger.warn(
+      `[requestPasswordReset] Tenant config blocked domain [Email: ${email}] [IP: ${req.ip}]`,
+    );
+    return makeDomainDeniedError();
```
[the remark this answers is not in the document]
Comment (danny-avila) [reply api/server/services/AuthService.js]: Good catch — fixed in 7b28b0ed8. The tenant domain block now returns the same generic "If an account with that email exists..." response instead of an Error, so the HTTP status is 200 in both the tenant-blocked and user-not-found paths. This prevents user-enumeration (and tenant-membership inference) via status code comparison.

The base-config fast-fail (pre-findUser) intentionally stays as an Error/400 since it fires before any user lookup and only reveals global domain policy, not per-user state.
